web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/AnnotationsApiController.java

```diff
@@ -761,170 +612,9 @@ private IndicatorQueryResp getIndicatorQueryFromHGVSg(
-            }
-        }
-        Integer index1 = name1.indexOf(this.keyword);
-        Integer index2 = name2.indexOf(this.keyword);
-        if (index1.equals(index2)) {
-            return compareLevel(i1, i2, name1, name2);
-        } else {
-            if (index1.equals(-1))
-                return 1;
-            if (index2.equals(-1))
-                return -1;
-            return compareLevel(i1, i2, name1, name2);
-        }
-
-    }
-
-    private Integer compareLevel(IndicatorQueryResp i1, IndicatorQueryResp i2, String name1, String name2) {
-        // Compare therapeutic levels
-        LevelOfEvidence i1Level = i1.getHighestSensitiveLevel();
-        LevelOfEvidence i2Level = i2.getHighestSensitiveLevel();
-        if (i1Level == null) {
-            i1Level = i1.getHighestResistanceLevel();
-        }
-        if (i2Level == null) {
-            i2Level = i2.getHighestResistanceLevel();
-        }
-        Integer result = LevelUtils.compareLevel(i1Level, i2Level, LevelUtils.getIndexedTherapeuticLevels());
-        if (result == 0) {
-            // Compare diagnostic level
-            result = LevelUtils.compareLevel(i1.getHighestDiagnosticImplicationLevel(), i2.getHighestDiagnosticImplicationLevel(), LevelUtils.getIndexedDiagnosticLevels());
-            if (result == 0) {
-                result = LevelUtils.compareLevel(i1.getHighestPrognosticImplicationLevel(), i2.getHighestPrognosticImplicationLevel(), LevelUtils.getIndexedPrognosticLevels());
-                if (result == 0) {
-                    result = LevelUtils.compareLevel(i1.getHighestFdaLevel(), i2.getHighestFdaLevel(), LevelUtils.getIndexedFdaLevels());
-                    if (result == 0) {
-                        //Compare Oncogenicity. Treat YES, LIKELY as the same
-                        Oncogenicity o1 = Oncogenicity.getByEffect(i1.getOncogenic());
-                        Oncogenicity o2 = Oncogenicity.getByEffect(i2.getOncogenic());
-                        if (o1 != null && o1.equals(Oncogenicity.LIKELY)) {
-                            o1 = Oncogenicity.YES;
-                        }
-                        if (o2 != null && o2.equals(Oncogenicity.LIKELY)) {
-                            o2 = Oncogenicity.YES;
-                        }
-                        result = MainUtils.compareOncogenicity(o1, o2, true);
-                        if (result == 0) {
-                            // Compare alteration name
-                            if (i1 == null || StringUtils.isNotEmpty(i1.getQuery().getAlteration())) {
-                                return 1;
-                            }
-                            if (i2 == null || StringUtils.isNotEmpty(i2.getQuery().getAlteration())) {
-                                return -1;
-                            }
-                            name1 = i1.getQuery().getAlteration().toLowerCase();
-                            name2 = i2.getQuery().getAlteration().toLowerCase();
```

Comment:
@calvinlu3 I removed these two lines to use from the parameters. Other changes are directly moved to Utils.